### PR TITLE
Creates an invoice entity to pay invoices

### PR DESCRIPTION
**Pull request by mikestone14**
Related to [DISP-4814](https://dispatchme.atlassian.net/browse/DISP-4814)

* Creates an `invoice` entity
* Submits invoice payment requests as authenticated `POST` requests to `/v1/invoices/:id/pay`
* Submits invoice payment requests with `amount` and `payment_method` params
* Defaults `payment_method` to `card`

### CODE REVIEW

**Comment by deadlocked247** [comment]
looks good to me but try to get another person to check it out before we merge it in/ publish to npm

File: lib/endpoints.js [hunk 1]
```diff
@@ -7,6 +7,7 @@ export const BILLING_ITEMS = '/v2/billing_items';
 export const BRANDS = '/v1/brands';
 export const CUSTOMERS = '/v1/customers';
 export const HYBRID_JOBS = '/v1/hybrid_jobs';
+export const INVOICES = '/v1/invoices';
```
Comment (derekcaneja) [inline lib/endpoints.js]: The endpoint for invoices should be `/v2/billing_documents`

File: lib/entities/invoice.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+const invoiceMethods = (client) => {
+  return (id) => {
+    return {
+      pay: ({ amount, cardToken, paymentMethod = 'card' }) => {
+        const formData = {
+          amount,
+          card_token: cardToken,
+          payment_method: paymentMethod,
+        };
+        const paymentEndpoint = `${INVOICES}/${id}/pay`;
```
Comment (derekcaneja) [inline lib/entities/invoice.js]: The payment endpoint should be `/v1/invoices/:id/payments`

File: lib/endpoints.js (hunk 1 above)
Comment (mikestone14) [inline lib/endpoints.js]: @derekcaneja this is only used to create the `paymentEndpoint`. Do you think it's a better idea to remove this line and create the endpoint inside the invoice entity file?

File: lib/entities/invoice.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+const invoiceMethods = (client) => {
+  return (id) => {
+    return {
+      pay: ({ amount, cardToken, paymentMethod = 'card' }) => {
```
Comment (aldeed) [inline lib/entities/invoice.js]: I'm not sure if defaulting the payment method is a good idea. Probably best to make them pass it explicitly.

Comment (aldeed) [inline lib/entities/invoice.js]: Or you could set it to "card" only if there's a cardToken

**Pull request merged**